portal/app/[locale]/bitcoin-yield/_hooks/useDeposit.ts

```diff
@@ -0,0 +1,191 @@
+      })
+
+      emitter.on('pre-approve', function () {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.APPROVAL_TX_PENDING,
+        })
+      })
+
+      emitter.on('user-signed-approval', function (approvalTxHash) {
+        updateBitcoinYieldOperation({
+          approvalTxHash,
+          status: BitcoinYieldDepositStatus.APPROVAL_TX_PENDING,
+        })
+      })
+
+      emitter.on('approve-transaction-succeeded', function (receipt) {
+        updateNativeBalanceAfterFees(receipt)
+        queryClient.invalidateQueries({ queryKey: allowanceQueryKey })
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.APPROVAL_TX_COMPLETED,
+        })
+      })
+
+      emitter.on('approve-transaction-reverted', function (receipt) {
+        updateNativeBalanceAfterFees(receipt)
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.APPROVAL_TX_FAILED,
+        })
+      })
+
+      emitter.on('user-signing-approval-error', function () {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.APPROVAL_TX_FAILED,
+        })
+      })
+
+      emitter.on('user-signed-deposit', function (transactionHash) {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.DEPOSIT_TX_PENDING,
+          transactionHash,
+        })
+      })
+
+      emitter.on('user-signing-deposit-error', function () {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.DEPOSIT_TX_FAILED,
+        })
+      })
+
+      emitter.on('deposit-transaction-succeeded', function (receipt) {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.DEPOSIT_TX_CONFIRMED,
+        })
+
+        // Update balances
+        updateNativeBalanceAfterFees(receipt)
+
+        // Update the user token balance
+        queryClient.setQueryData(
+          tokenBalanceQueryKey,
+          (old: bigint) => old - amount,
+        )
+
+        // Update pool deposits (add deposited amount)
+        queryClient.setQueryData(
+          poolDepositsQueryKey,
+          (old: bigint) => old + amount,
+        )
+
+        // Update user pool balance (add deposited amount to assets
+        // converted from vault shares)
+        queryClient.setQueryData(
+          userPoolBalanceQueryKey,
+          (old: bigint) => old + amount,
+        )
+      })
+
+      emitter.on('deposit-transaction-reverted', function (receipt) {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.DEPOSIT_TX_FAILED,
+        })
+
+        // Although the transaction was reverted, the gas was paid.
+        updateNativeBalanceAfterFees(receipt)
+      })
+
+      emitter.on('deposit-failed-validation', function () {
+        updateBitcoinYieldOperation({
+          status: BitcoinYieldDepositStatus.DEPOSIT_TX_FAILED,
+        })
+      })
+
+      on?.(emitter)
```

Comment:
?

Reply:
can you be more specific? I don't read minds (yet)

Reply:
Disregard!